### PR TITLE
Add table aws_securitylake_data_lake closes #1441

### LINKED ISSUE #1441: Add table aws_securitylake_data_lake
**References**
https://github.com/aws/aws-sdk-go-v2/blob/main/service/securitylake/api_op_GetDatalake.go#L16

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  encryption_key,
  replication_role_arn,
  s3_bucket_arn,
  status
from
  aws_securitylake_data_lake;
+----------------+-------------------------------------------------------------------------+------------------------------------------------------------------------------+-----------+
| encryption_key | replication_role_arn                                                    | s3_bucket_arn                                                                | status    |
+----------------+-------------------------------------------------------------------------+------------------------------------------------------------------------------+-----------+
| S3_MANAGED_KEY | <null>                                                                  | arn:aws:s3:::aws-security-data-lake-eu-west-1-e1mtg9v0rolvlglniawrhjf4ialrti | COMPLETED |
| S3_MANAGED_KEY | <null>                                                                  | arn:aws:s3:::aws-security-data-lake-us-east-1-qeydzsin6buhruyudnvt7gnkpmxck3 | COMPLETED |
| S3_MANAGED_KEY | arn:aws:iam::938702159474:role/SecurityLake_S3ReplicationRole_us_east_1 | arn:aws:s3:::aws-security-data-lake-us-east-2-7pthxugfyv6u5uzyd6f3qt0tgd1mlu | COMPLETED |
| S3_MANAGED_KEY | <null>                                                                  | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | COMPLETED |
| S3_MANAGED_KEY | <null>                                                                  | arn:aws:s3:::aws-security-data-lake-ap-southeast-2-4i3pnujz7bpnx72x6oheacrcx | COMPLETED |

> select
  b.name as bucket_name,
  l.s3_bucket_arn,
  b.creation_date,
  b.bucket_policy_is_public,
  b.versioning_enabled,
  b.block_public_acls
from
  aws_securitylake_data_lake as l,
  aws_s3_bucket as b
where
  l.s3_bucket_arn = b.arn;
+-----------------------------------------------------------------+------------------------------------------------------------------------------+---------------------------+-------------------------+--------------------+-------------------+
| bucket_name                                                     | s3_bucket_arn                                                                | creation_date             | bucket_policy_is_public | versioning_enabled | block_public_acls |
+-----------------------------------------------------------------+------------------------------------------------------------------------------+---------------------------+-------------------------+--------------------+-------------------+
| aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | 2022-12-01T15:37:10+05:30 | false                   | false              | true              |
| aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | 2022-12-01T15:37:10+05:30 | false                   | false              | true              |
| aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | 2022-12-01T15:37:10+05:30 | false                   | false              | true              |
| aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | 2022-12-01T15:37:10+05:30 | false                   | false              | true              |
| aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | 2022-12-01T15:37:10+05:30 | false                   | false              | true              |
| aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | 2022-12-01T15:37:10+05:30 | false                   | false              | true              |
| aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | 2022-12-01T15:37:10+05:30 | false                   | false              | true              |
| aws-security-data-lake-eu-central-1-5kzifb2pzuzw7kgalmmvfsh9pxq | arn:aws:s3:::aws-security-data-lake-eu-central-1-5kzifb2pzuzw7kgalmmvfsh9pxq | 2022-12-01T15:36:49+05:30 | false                   | false              | true              |
| aws-security-data-lake-eu-central-1-5kzifb2pzuzw7kgalmmvfsh9pxq | arn:aws:s3:::aws-security-data-lake-eu-central-1-5kzifb2pzuzw7kgalmmvfsh9pxq | 2022-12-01T15:36:49+05:30 | false                   | false              | true              |
| aws-security-data-lake-eu-central-1-5kzifb2pzuzw7kgalmmvfsh9pxq | arn:aws:s3:::aws-security-data-lake-eu-central-1-5kzifb2pzuzw7kgalmmvfsh9pxq | 2022-12-01T15:36:49+05:30 | false                   | false              | true       

> select
  distinct b.name as bucket_name,
  l.s3_bucket_arn,
  b.creation_date,
  b.bucket_policy_is_public,
  b.versioning_enabled,
  b.block_public_acls
from
  aws_securitylake_data_lake as l,
  aws_s3_bucket as b
where
  l.s3_bucket_arn = b.arn;
+-----------------------------------------------------------------+------------------------------------------------------------------------------+---------------------------+-------------------------+--------------------+-------------------+
| bucket_name                                                     | s3_bucket_arn                                                                | creation_date             | bucket_policy_is_public | versioning_enabled | block_public_acls |
+-----------------------------------------------------------------+------------------------------------------------------------------------------+---------------------------+-------------------------+--------------------+-------------------+
| aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | arn:aws:s3:::aws-security-data-lake-ap-northeast-1-yd5aigzmgod7uadzgou2d2otb | 2022-12-01T15:37:10+05:30 | false                   | false              | true              |
| aws-security-data-lake-ap-southeast-2-4i3pnujz7bpnx72x6oheacrcx | arn:aws:s3:::aws-security-data-lake-ap-southeast-2-4i3pnujz7bpnx72x6oheacrcx | 2022-12-01T15:37:29+05:30 | false                   | false              | true              |
| aws-security-data-lake-eu-central-1-5kzifb2pzuzw7kgalmmvfsh9pxq | arn:aws:s3:::aws-security-data-lake-eu-central-1-5kzifb2pzuzw7kgalmmvfsh9pxq | 2022-12-01T15:36:49+05:30 | false                   | false              | true              |
| aws-security-data-lake-eu-west-1-e1mtg9v0rolvlglniawrhjf4ialrti | arn:aws:s3:::aws-security-data-lake-eu-west-1-e1mtg9v0rolvlglniawrhjf4ialrti | 2022-12-01T15:36:40+05:30 | false                   | false              | true              |
| aws-security-data-lake-us-east-1-qeydzsin6buhruyudnvt7gnkpmxck3 | arn:aws:s3:::aws-security-data-lake-us-east-1-qeydzsin6buhruyudnvt7gnkpmxck3 | 2022-12-01T15:36:58+05:30 | false                   | true               | true              |
| aws-security-data-lake-us-east-2-7pthxugfyv6u5uzyd6f3qt0tgd1mlu | arn:aws:s3:::aws-security-data-lake-us-east-2-7pthxugfyv6u5uzyd6f3qt0tgd1mlu | 2022-12-01T15:37:23+05:30 | false                   | true               | true              |
| aws-security-data-lake-us-west-2-xnssh4doirrzwpxmcfxy3vrbsbbu4k | arn:aws:s3:::aws-security-data-lake-us-west-2-xnssh4doirrzwpxmcfxy3vrbsbbu4k | 2022-12-01T15:37:02+05:30 | false                   | false              | true              |
+-----------------------------------------------------------------+------------------------------------------------------------------------------+---------------------------+-------------------------+--------------------+-------------------+

```
</details>

### CODE REVIEW

File: aws/table_aws_securitylake_data_lake.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "encryption_key",
+				Description: "The type of encryption key used by Security Lake to encrypt the lake configuration.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "replication_role_arn",
+				Description: "Replication settings for the Amazon S3 buckets. This parameter uses the IAM role created by you that is managed by Security Lake, to ensure the replication setting is correct.",
```
Comment (misraved) [inline aws/table_aws_securitylake_data_lake.go:30]:
```suggestion
				Description: "This parameter uses the IAM role created by you that is managed by Security Lake, to ensure the replication setting is correct.",
```

File: aws/table_aws_securitylake_data_lake.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "s3_bucket_arn",
+				Description: "Amazon Resource Names (ARNs) uniquely identify Amazon Web Services resources.",
```
Comment (misraved) [inline aws/table_aws_securitylake_data_lake.go:35]: Is the description correct?

File: aws/table_aws_securitylake_data_lake.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+//// LIST FUNCTION
+
+func getSecurityLakeDataLake(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	// Create Client
+	svc, err := SecurityLakeClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_securitylake_data_lake.listSecurityLakeDataLakes", "client_error", err)
```
Comment (misraved) [inline aws/table_aws_securitylake_data_lake.go:72]:
```suggestion
		plugin.Logger(ctx).Error("aws_securitylake_data_lake. getSecurityLakeDataLake", "client_error", err)
```

File: aws/table_aws_securitylake_data_lake.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	if svc == nil {
+		// Unsupported region check
+		return nil, nil
+	}
+
+	input := &securitylake.GetDatalakeInput{}
+
+	resp, err := svc.GetDatalake(ctx, input)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_securitylake_data_lake.listSecurityLakeDataLakes", "api_error", err)
```
Comment (misraved) [inline aws/table_aws_securitylake_data_lake.go:84]:
```suggestion
		plugin.Logger(ctx).Error("aws_securitylake_data_lake. getSecurityLakeDataLake", "api_error", err)
```

File: docs/tables/aws_securitylake_data_lake.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+select
+  encryption_key,
+  replication_role_arn,
+  s3_bucket_arn,
+  status
+from
+  aws_securitylake_data_lake;
+```
+
+### Get S3 bucket details for each data lake
```
Comment (misraved) [inline docs/tables/aws_securitylake_data_lake.md:19]:
```suggestion
### Get S3 bucket details of each data lake
```

File: docs/tables/aws_securitylake_data_lake.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  b.versioning_enabled,
+  b.block_public_acls
+from
+  aws_securitylake_data_lake as l,
+  aws_s3_bucket as b
+where
+  l.s3_bucket_arn = b.arn;
+```
+
+### Get retention setting details of data lakes
```
Comment (misraved) [inline docs/tables/aws_securitylake_data_lake.md:36]:
```suggestion
### Get retention setting details of data lake
```

File: docs/tables/aws_securitylake_data_lake.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  l.s3_bucket_arn,
+  l.status,
+  r ->> 'RetentionPeriod' as retention_period,
+  r ->> 'StorageClass' as storage_class
+from
+  aws_securitylake_data_lake as l,
+  jsonb_array_elements(retention_settings) as r;
+```
+
+### List data lakes that are not compleated
```
Comment (misraved) [inline docs/tables/aws_securitylake_data_lake.md:51]:
```suggestion
### List data lakes that are not completed
```

**Review by misraved — COMMENTED** [review]
@ParthaI The AWS document suggests that we might get `ResourceNotFoundException` when we try to access data lakes. Should we handle this in the `IgnoreConfig` section?

Please take a look at the other review comments. Thanks!!

File: docs/tables/aws_securitylake_data_lake.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+```sql
+select
+  encryption_key,
+  replication_role_arn,
+  s3_bucket_arn,
+  status
+from
+  aws_securitylake_data_lake
+where
+  status <> 'COMPLETED';
```
Comment (misraved) [inline docs/tables/aws_securitylake_data_lake.md:62]:
```suggestion
  status = 'PENDING';
```

File: aws/table_aws_securitylake_data_lake.go (hunk 2 above)
Comment (ParthaI) [reply aws/table_aws_securitylake_data_lake.go:35]: The same description is mentioned in doc.
https://docs.aws.amazon.com/security-lake/latest/APIReference/API_LakeConfigurationResponse.html

**Review by misraved — APPROVED** [review]

**Pull request merged**